### PR TITLE
ENT-12366: External verifier now sets appclassloader to legacy contra…

### PR DESCRIPTION
**Pull request by adelel1**
ENT-12366: External verifier now sets appclassloader to legacy contracts directory instead of the cordapps directory.

### CODE REVIEW

File: verifier/src/main/kotlin/net/corda/verifier/ExternalVerifier.kt [hunk 1]
```diff
@@ -86,7 +86,7 @@ class ExternalVerifier(private val baseDirectory: Path, private val channel: Soc
         initialise()
         while (true) {
             val request = channel.readCordaSerializable(VerificationRequest::class)
-            log.debug { "Received $request" }
+            safeLog { log.debug { "Received $request" }}
```
Comment (rick-r3) [inline verifier/src/main/kotlin/net/corda/verifier/ExternalVerifier.kt]: I think we should change the `toString()` of `VerificationRequest` to just output the transaction ID, not the transaction.  I think possible prematurely deserialising the transaction might have negative effects beyond just causing class loader problems.

File: node/src/main/kotlin/net/corda/node/internal/cordapp/JarScanningCordappLoader.kt [hunk 2]
```diff
@@ -220,9 +220,11 @@ class JarScanningCordappLoader(private val cordappJars: Set<Path>,
         private fun checkSignersMatch(legacyCordapp: CordappImpl, nonLegacyCordapp: CordappImpl) {
             val legacySigners = legacyCordapp.jarPath.openStream().let(::JarInputStream).use(JarSignatureCollector::collectSigners)
             val nonLegacySigners = nonLegacyCordapp.jarPath.openStream().let(::JarInputStream).use(JarSignatureCollector::collectSigners)
-            check(rotatedKeys.canBeTransitioned(legacySigners, nonLegacySigners)) {
-                "Newer contract CorDapp '${nonLegacyCordapp.jarFile}' signers do not match legacy contract CorDapp " +
-                        "'${legacyCordapp.jarFile}' signers."
+            if (legacySigners.size > 0 || nonLegacySigners.size > 0) {
```
Comment (rick-r3) [inline node/src/main/kotlin/net/corda/node/internal/cordapp/JarScanningCordappLoader.kt]: Use `isNotEmpty()` ?

File: verifier/src/main/kotlin/net/corda/verifier/ExternalVerifier.kt (hunk 1 above)
Comment (adelel1) [reply verifier/src/main/kotlin/net/corda/verifier/ExternalVerifier.kt]: Done

File: node/src/main/kotlin/net/corda/node/internal/cordapp/JarScanningCordappLoader.kt (hunk 2 above)
Comment (adelel1) [reply node/src/main/kotlin/net/corda/node/internal/cordapp/JarScanningCordappLoader.kt]: Done

**Review by rick-r3 — APPROVED** [review]

**Pull request merged**